src/utils/components/PolicyForm/PolicyInterfaceExpandable.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useRef, useState } from 'react';
+import { Updater } from 'use-immer';
+
+import { Button, FormFieldGroupExpandable, FormFieldGroupHeader } from '@patternfly/react-core';
+import { MinusCircleIcon } from '@patternfly/react-icons';
+import { NodeNetworkConfigurationInterface, V1NodeNetworkConfigurationPolicy } from '@types';
+import { useNMStateTranslation } from '@utils/hooks/useNMStateTranslation';
+
+import DeleteInterfaceModal from './DeleteInterfaceModal';
+import PolicyInterface, { onInterfaceChangeType } from './PolicyInterface';
+import { getExpandableTitle } from './utils';
+
+type PolicyInterfacesExpandableProps = {
+  policy: V1NodeNetworkConfigurationPolicy;
+  setPolicy: Updater<V1NodeNetworkConfigurationPolicy>;
+  createForm?: boolean;
+};
+
+const PolicyInterfacesExpandable: React.FC<PolicyInterfacesExpandableProps> = ({
+  policy,
+  setPolicy,
+  createForm,
+}) => {
+  const createdPolicy = useRef(createForm ? undefined : policy);
+  const [interfaceToDelete, setInterfaceToDelete] = useState<NodeNetworkConfigurationInterface>();
+  const { t } = useNMStateTranslation();
+
+  const removeInterface = (interfaceIndex: number) => {
+    const createdInterfacesNames = createdPolicy?.current?.spec?.desiredState?.interfaces?.map(
+      (inFace) => inFace?.name,
+    );
+
+    if (
+      createdInterfacesNames?.includes(policy.spec.desiredState.interfaces[interfaceIndex].name)
```

Comment:
Maybe better `policy?.spec?.desiredState?.interfaces[interfaceIndex]?.name` ? Same as used in the lines above. WDYT? Just in case.. ;)